tools/sql_extraction/src/main/kotlin/Cli.kt

```diff
@@ -0,0 +1,30 @@
+package com.google.cloud.sqlecosystem.sqlextraction
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.multiple
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.path
+import mu.KotlinLogging
+import java.nio.file.Path
+
+private val logger = KotlinLogging.logger { }
+
+fun main(args: Array<String>) = Cli().main(args)
+
+class Cli : CliktCommand(name = "sql_extraction") {
+    val recursive: Boolean by option("-r", "--recursive", help = "scan files in subdirectories recursively").flag()
+    val filePaths: List<Path> by argument(help = "file and directory paths to code").path(
+        mustExist = true,
+        canBeDir = true,
+        canBeFile = true
+    ).multiple()
+
+    override fun run() {
+        logger.debug("Starting SQL Extraction from command line")
+
+        val files = FileListExpander().expandAndFilter(filePaths, recursive)
+        logger.debug { "Files to analyze: $files" }
+    }
+}
```

Comment:
Please print out the usage when -h is specified. In the usage, please also include several sample usages. 

It seems right now the program doesn't support file patterns like "*.cs" and it's hard-coded to look for java files?

Reply:
Good idea for printing sample usages!

Currently, only java files are included since I wanted to focus on it first. But maybe it's a better idea to accept all file types here and filter them later based on which language frontends are implemented. This should also bring back the reason to support the --filetype flag that you mentioned.

Reply:
you don't have to add support for non-java files for now.